contentful_migrations/migrations/1662646314014-new-compose--drop-old-compose-page-data-and-content-type.cjs

```diff
@@ -0,0 +1,71 @@
+/* eslint-env node */
+const { withHelpers } = require('@jungvonmatt/contentful-migrations');
+
+// DEPENDS on 1661197009323--new-compose--add-fields.cjs and its prerequisite migration
+
+/**
+ * Contentful migration
+ * API: https://github.com/contentful/contentful-migration
+ * Editor Interfaces: https://www.contentful.com/developers/docs/extensibility/app-framework/editor-interfaces/
+ */
+
+const sleep = (milliseconds) =>
+    new Promise((res) => setTimeout(res, milliseconds));
+
+module.exports = withHelpers(async function (migration, context, helpers) {
+    const LEGACY_COMPOSE_PAGE = 'page';
+
+    // // 0. Get all Compose:Page entries
+    allEntries = await context.makeRequest({
+        method: 'get',
+        url: '/entries'
+    });
+    const legacyComposePageEntries = allEntries.items.filter(
+        (entry) => entry.sys.contentType.sys.id == LEGACY_COMPOSE_PAGE
+    );
+
+    // 1. Unpublish all Compose:Page entries
+    await legacyComposePageEntries.forEach((composePageEntry) => {
+        if (composePageEntry.sys.publishedVersion) {
+            console.log(`Unpublishing Compose:Page ${composePageEntry.sys.id}`);
+            const resp = context.makeRequest({
+                method: 'delete',
+                url: `/entries/${composePageEntry.sys.id}/published`,
+                headers: {
+                    'X-Contentful-Version':
+                        composePageEntry.sys.publishedVersion
+                }
+            });
+            console.log('Compose:Page unpublishing response:', resp);
+        } else {
+            console.log(
+                `Compose:Page ${composePageEntry.sys.id} was not Published, so skipping`
+            );
+        }
+    });
+
+    // Give Contentful a moment to breathe before moving on
```

Comment:
**question**: were we running into rate limit errors? what necessitated the sleep step?

Reply:
Yes we were. I'll update the comment to be less hand-wavy!